modules/caddyhttp/metrics.go

```diff
@@ -181,3 +182,28 @@ func computeApproximateRequestSize(r *http.Request) int {
 	}
 	return s
 }
+
+// taken from https://github.com/prometheus/client_golang/blob/9ac0bad6062845400019e68d4fa0bf0384b5b6f6/prometheus/promhttp/instrument_server.go#L322
+// and modified to uppercase the method.
+func sanitizeMethod(m string) string {
```

Comment:
Simiolarly, just use `strings.ToUpper()`. Probably don't even need a separate function for this (although Go's compiler will inline it, I bet).